WorkWithThreadPool/WorkWithThreadPool/MyThreadPool.cs

```diff
@@ -0,0 +1,204 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+
+namespace WorkWithThreadPool
+{   
+    /// <summary>
+    /// класс для параллельных исполнений задач
+    /// </summary>
+    public class MyThreadPool
+    {
+        private ConcurrentQueue<Action> tasks;
+
+        private Thread[] threads;
+
+        private AutoResetEvent threadPoolEvent;
+
+        private CancellationTokenSource _cancellationToken;
+
+        private int tasksCount; 
+        
+        
+        /// <summary>
+        /// Класс для параллельных вычислений
+        /// </summary>
+        private class MyTask<T> : IMyTask<T>
+        {
+            private T _result;
+    
+            private MyThreadPool _threadPool;
+    
+            private bool _isCompleted;
+            
+            private Func<T> task;
+    
+            private ConcurrentQueue<Action> _continueTasks;
+
+            private AggregateException _exception;
+
+            private ManualResetEvent resultCalculated;
+
+            public MyTask(Func<T> task, MyThreadPool pool)
+            {
+                this.task = task;
+                _threadPool = pool;
+                _continueTasks = new ConcurrentQueue<Action>();
+                resultCalculated = new ManualResetEvent(false);
+            }
+            
+            /// <inheritdoc />
+            public T Result
+            {
+                get
+                {
+                    resultCalculated.WaitOne();
+                    if (_exception != null)
+                    {
+                        throw _exception;
+                    }
+                    return _result;
+                }
+            }
+            
+            /// <inheritdoc />
+            public bool IsCompleted { get => _isCompleted; }
```

Comment:
```suggestion
            public bool IsCompleted => _isCompleted;
```
Это то же самое, но писать меньше